### ISSUE TITLE
MySQL BIT data type causes strange behavior

### ISSUE THREAD

**Issue by dcaravana** [balderdashy/sails-mysql#178]
Values of `BIT` data type are returned as:
```
...
    "isVerified": {
      "0": 0,
      "length": 1,
      "parent": [
        1,
        0,
        0,
        ... lots of garbage ...
        1,
        0,
        49,
        0
      ],
      "offset": 4304
    },
...
```
independently of the corresponding model type (either `boolean` or `integer`).

If e.g. `TINYINT(1)` is used, it works as expected instead.

Using MySQL 5.5.29, sails 0.10.5 and sails-mysql 0.10.8.

**Comment by dcaravana** [comment]
Is #128 related?

**Comment by dcaravana** [comment]
I went deeper inside the issue: the `BIT` data type is not only 1 bit but can be [up to 64 bits](http://dev.mysql.com/doc/refman/5.0/en/bit-type.html) which mysql node.js driver [packs in a buffer](https://www.npmjs.com/package/mysql#buffer) of max 8 bytes (integers), so mapping a `boolean` to a `BIT` is basically wrong from a semantic standpoint (and this is why `BOOL` data type is a [synonym for `TINYINT(1)` to MySQL](http://dev.mysql.com/doc/refman/5.6/en/numeric-type-overview.html)).

And this is also why #128 is potentially wrong (or not if we want to assume that a `boolean` corresponds to a `BIT(1)`).

Closing this issue as it can be considered invalid.
